xmtp/src/builder.rs

```diff
@@ -33,14 +36,47 @@ where
         self
     }
 
+    pub fn find_or_create_account(mut self, wallet_address: String) -> Result<Self, String> {
+        let key = get_account_storage_key(wallet_address);
+        let persistence = self.persistence.as_mut().ok_or_else(|| {
```

Comment:
super nit: the mut ref persistence is only needed for the write operation in the 2nd match arm. 

Since the object is owned you could only take the &mut if needed which would relax constraints from the borrow checker. Not super applicable in this code, but since mut refs are exclusive, this could cause borrow errors in the future/other cirumstances